### PR TITLE
Feature/signin

### PR DESCRIPTION
**Pull request by irtazazulfiqar**

### CODE REVIEW

File: frontend/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_BACKEND_BASE_URL=http://127.0.0.1:8000
```
Comment (m-shayan-ahmad) [inline frontend/.env:1]: Should this be pushed?

Comment (m-shayan-ahmad) [reply frontend/.env:1]: This is me repeating the same suggestion.

File: frontend/src/pages/SignUp.js [hunk 2]
```diff
@@ -142,6 +142,13 @@ function SignUp() {
           >
             Sign Up
           </Button>
+          {message && <Typography color="success.main">{message}</Typography>}
+          {/* Display general errors if any */}
+          {Object.keys(errors).length > 0 && !Object.values(errors).some(errs => errs.length > 0) && (
+            <Typography color="error.main">
+              {Object.values(errors).flat().join(', ')}
+            </Typography>
+          )}
```
Comment (m-shayan-ahmad) [inline frontend/src/pages/SignUp.js:151]: 👍

**Review by m-shayan-ahmad — CHANGES_REQUESTED** [review]

File: frontend/.env
Comment (m-shayan-ahmad) [inline frontend/.env:1]: Did we add it in the gitignore?

Comment (irtazazulfiqar) [reply frontend/.env:1]: `.env` was already mentioned in the `.gitignore` but due to intermingling of PR's it was somehow merged up and committed, so now i just deleted `.env` and made a new one and it is now following `.gitignore` file and we can't see `.env` in git status command now.

**Review by m-shayan-ahmad — APPROVED** [review]

**Pull request merged**